components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -3081,13 +3089,42 @@ private FederatedAuthenticator createFederatedAuthenticator(String authenticator
                     federatedAuthenticator.setTags(Arrays.asList(tags));
                 }
             }
-            List<org.wso2.carbon.identity.api.server.idp.v1.model.Property> properties =
-                    Arrays.stream(config.getProperties()).map(propertyToExternal).collect(Collectors.toList());
-            federatedAuthenticator.setProperties(properties);
+
+            if (DefinedByType.SYSTEM == config.getDefinedByType()) {
+                federatedAuthenticator.setDefinedBy(FederatedAuthenticator.DefinedByEnum.SYSTEM);
+                List<org.wso2.carbon.identity.api.server.idp.v1.model.Property> properties =
+                        Arrays.stream(config.getProperties()).map(propertyToExternal).collect(Collectors.toList());
+                federatedAuthenticator.setProperties(properties);
+            } else {
+                federatedAuthenticator.setDefinedBy(FederatedAuthenticator.DefinedByEnum.USER);
+                resolveEndpointConfiguration(federatedAuthenticator, config);
+            }
+
         }
         return federatedAuthenticator;
     }
```

Comment:
Doesn't this method now comes from the factory for system defined and user defined authenticators ?

Reply:
In the factory, we are converting Auth API model -> Service layer fed model. Here what happens is Service layer fed model endpoint -> Auth API model enpoint